Make slurm execute script customizable

linsword13 opened the pull request:
This is done by making the template source path pointing to a
workflow_manager_variable.

The major change is around extending `register_template`'s `src_path`
prop (renamed from `src_name`) to allow it to point to a user-specified
path.

Another change related to template rendering is to define `template_render_vars` as reserved variables.

linsword13 committed:
Make slurm execute script customizable

This is done by making the template source path pointing to a
workflow_manager_variable.

The major change is around extending `register_template`'s `src_path`
prop (renamed from `src_name`) to allow it to point to a user-specified
path.

Another change is to point the destination of the slurm execute script
to `execute_experiment`. The intent is to make the ux similar to
workspaces that don't use workflow manager, in that manually changing
the rendered `execute_experiment` in the experiment_run_dir takes
effect.

linsword13 committed:
Add in template tests to cover the path expansion

douglasjacobsen commented on var/ramble/repos/builtin/workflow_managers/slurm/workflow_manager.py:
```diff
@@ -83,35 +83,44 @@ def __init__(self, file_path):
+    workflow_manager_variable(
+        name="slurm_execute_template_path",
+        default="slurm_execute_experiment.tpl",
+        description="Path to the custom template (under workspace_config dir) for generating "
```
I think `under workspace_config dir` is a bit misleading here. The default is that it's under the configs dir, but allowing other workspace path variables (or an absolute path) this file can be anywhere on a file system, right?

linsword13 replied:
Indeed, the description was wrong. Updated.

douglasjacobsen commented on var/ramble/repos/builtin/workflow_managers/slurm/workflow_manager.py:
```diff
@@ -83,35 +83,44 @@ def __init__(self, file_path):
-        src_name="slurm_execute_experiment.tpl",
-        dest_path="slurm_execute_experiment",
+        src_path="{slurm_execute_template_path}",
+        dest_path="execute_experiment",
```
I am not sure how I feel about this hijacking `execute_experiment`. But I do understand why you did it (partly because I complained about how it felt confusing).

It's also kind of weird to have a template file not end up rendering as a template (I'll send you more about this in work chat).

linsword13 replied:
Ack. Reverted back to the `slurm_execute_experiment`.

linsword13 committed:
Define `template_render_vars` as reserved variables

This allows other templates (not registered via `register_template`) to
be able to use these variables.

linsword13 committed:
Revert back to use "slurm_execute_experiment" as the rendered script
destination

linsword13 committed:
Fix a wrong comment

linsword13 committed:
Include in searched paths in error message

linsword13 committed:
Rename the slurm batch script to slurm_experiment_sbatch

linsword13 committed:
Make register_template's `dest_path` optional

linsword13 committed:
Use the same input file name for the rendered slurm execute script
